src/consumers.ts

```diff
@@ -124,36 +147,56 @@ class EmptyResultEngine extends AnalysisConsumer implements DiagnosticProducer
     }   
 }
 
+/* DocumentUri as a URI object to pass URLs */
+let targerLink : DocumentUri;
+
 /* Report CVEs in found dependencies */
 class SecurityEngine extends AnalysisConsumer implements DiagnosticProducer
 {
     constructor(public context: IDependency, config: any) {
         super(config);
-        this.binding = {path: ['result', 'recommendation', 'component-analyses', 'cve']};
+        this.binding = {path: ['component_analyses', 'vulnerability']};
         /* recommendation to use a different version */
-        this.changeToBinding = {path: ['result', 'recommendation', 'change_to']};
+        this.changeToBinding = {path: ['recommended_versions']};
+        /* snyk registration link */
+        this.regLinkBinding = {path: ['registration_link']};
+        /* Diagnostic message */
+        this.messageBinding = {path: ['message']};
+        /* Public and Private vulnerability count */
+        this.pubVulBinding = {path: ['known_security_vulnerability_count']};
+        this.pvtVulBinding = {path: ['security_advisory_count']};
     }
 
     produce(ctx: any): Diagnostic[] {
         if (this.item.length > 0) {
-            let cveList = [];
-            for (let cve of this.item) {
-                cveList.push(cve['id'])
+            /* Counting total # of Public and Private Vulnerability */
+            VulPrivate += this.pvtVul;
+            VulPublic += this.pubVul;
+            /* Assign a string to a type DocumentUri */
+            targerLink = this.regLink;
```

Comment:
If it is for type conversion, then use
```
<DocumentUri> this.regLink
```
or
```
this.regLink as DocumentUri
```

Reply:
+1